### PR TITLE
Add plugin permission validation

### PR DESCRIPTION
**Pull request by rjernst**
Security manager policies within plugins currently can ask to grant any
permission (though we block some within the security manager itself at
runtime). Yet most of these permissions should never be necessary, and
some we would actively not want any plugins to be allowed to use. This
commit adds validation of plugins' policy files to restrict the
permissions allowed to be granted to a subset that is reasonable for
plugins to need. The allowed permissions are not ideal (still containing
things like suppressAccessChecks), but it is a step forward in defining
a stricter model for plugins that reduces the surface area of potential
abuse.

### CODE REVIEW

**Comment by rjernst** [comment]
@jasontedor This is ready for review

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/bootstrap/PolicyUtil.java [hunk 1]
```diff
@@ -23,31 +23,170 @@
...
+        // classes have their own meaning for some form of wildcard matching of the name, which we want to delegate
+        // to those permissions if possible.
+        Map<String, List<String>> classPermissions = Map.of(
+            URLPermission.class, ALLOW_ALL_NAMES,
+            DelegationPermission.class, ALLOW_ALL_NAMES,
+            ServicePermission.class, ALLOW_ALL_NAMES,
+            PrivateCredentialPermission.class, ALLOW_ALL_NAMES,
+            SQLPermission.class, List.of("callAbort", "setNetworkTimeout"),
+            ClassPermission.class, ALLOW_ALL_NAMES
+        ).entrySet().stream().collect(Collectors.toMap(e -> e.getKey().getCanonicalName(), Map.Entry::getValue));
```
Comment (jaymode) [inline server/src/main/java/org/elasticsearch/bootstrap/PolicyUtil.java:163]: Is there a reason we don't just build the map with the canonical class name?

```suggestion
        Map<String, List<String>> classPermissions = Map.of(
            URLPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            DelegationPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            ServicePermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            PrivateCredentialPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            SQLPermission.class.getCanonicalName(), List.of("callAbort", "setNetworkTimeout"),
            ClassPermission.class.getCanonicalName(), ALLOW_ALL_NAMES
        );
```

**Review by jaymode — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/bootstrap/PolicyUtil.java (hunk 1 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/bootstrap/PolicyUtil.java:163]: Just more cumbersome to add. I found it easier to do the common manipulation at the end on all the entries.

**Comment by fcofdez** [comment]
@rjernst I'm working on upgrading the azure repository SDK to a newer version and that new version relies on a Proxy
https://github.com/elastic/elasticsearch/pull/65140/files#diff-11d79e3153671e64610a6064df6c0bc159177e179a80a8bdaf0d8aff085b10a9R23 and this PR introduced a ban on that permission. Is there a workaround around this? Thanks!

**Pull request merged**